openedx/edx-ora2
# feat: resolve studio urls in frontend

**TL;DR -** There are a fair number of "unresolved" studio URLs present in production ORA prompt definitions. Rather than attempting to fix them, we've stopped them from being introduced in other fixes, and with this are attempting to resolve them as they are displayed.

JIRA: [AU-555](https://openedx.atlassian.net/browse/AU-555)

**What changed?**

- Added "base_asset_url" to response_mixin context which is inserted into the DOM as a data attribute in the response template
-  Created a Prompts class and Prompt js class to interface with all prompts / individual prompts respectively
- Added logic to ResponseView to, on load, attempt to resolve and re-display the prompt content with resolved links.

**Developer Checklist**


**Testing Instructions**

- Download this course ([course.tar.gz](https://github.com/openedx/edx-ora2/files/8352064/course.tar.gz)), and import it into your local Studio on master. It contains one ORA.
- Navigate to the ORA in the LMS and Studio and observe the different behaviors of links / images / media.
- Check out this branch, install-local, and then attach and restart both lms and studio.
- Navigate to the ORA in the LMS and Studio and observe the new behavior and how it affects links / images / media.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @openedx/content-aurora
